mule-extensions-api/src/test/java/org/mule/runtime/extension/internal/loader/enricher/ExtensionTypesDeclarationEnricherTestCase.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.loader.enricher;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singleton;
+import static java.util.Collections.singletonList;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.mule.metadata.api.annotation.TypeAnnotation;
+import org.mule.metadata.api.annotation.TypeIdAnnotation;
+import org.mule.metadata.api.builder.ObjectFieldTypeBuilder;
+import org.mule.metadata.api.builder.ObjectTypeBuilder;
+import org.mule.metadata.api.model.MetadataFormat;
+import org.mule.metadata.api.model.ObjectFieldType;
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.model.impl.DefaultObjectType;
+import org.mule.metadata.java.api.annotation.ClassInformationAnnotation;
+import org.mule.runtime.api.meta.model.SubTypesModel;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclarer;
+import org.mule.runtime.api.meta.model.declaration.fluent.OperationDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.OutputDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterGroupDeclaration;
+import org.mule.runtime.extension.api.declaration.type.annotation.ParameterDslAnnotation;
+import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class ExtensionTypesDeclarationEnricherTestCase {
+
+  @Test
+  public void subTypesNormalized() {
```

Comment:
are you testing normalization or flyWeight? Either add a comment explaining what normalization means or rename to a more meaningful name